### PR TITLE
Feature/script machine

### LINKED ISSUE #5: Support more OP_CODEs and Scripts
Refer to the following CoreBitcoin implementation
- [CoreBitcoin/BTCOpcode.h](https://github.com/oleganza/CoreBitcoin/blob/master/CoreBitcoin/BTCOpcode.h)
- [CoreBitcoin/BTCScript.h](https://github.com/oleganza/CoreBitcoin/blob/master/CoreBitcoin/BTCScript.h)
- [CoreBitcoin/BTCScriptMachine.h](https://github.com/oleganza/CoreBitcoin/blob/master/CoreBitcoin/BTCScriptMachine.h)

### PR DESCRIPTION
**Pull request by usatie**
Add ScriptMachine

closes #5

### CODE REVIEW

File: BitcoinKit/ScriptMachine.swift [hunk 1]
```diff
@@ -710,163 +710,163 @@ class ScriptMachine {
...
+        guard let tx = transaction else {
+            throw ScriptMachineError.error("transaction should not be nil.")
         }
 
-        print("STATUS", status)
-
-        guard status == 1 else {
-            return false
+        let sighash: Data = tx.signatureHash(for: utxoToSign, inputIndex: inputIndex, hashType: hashType)
+        if Crypto.verifySignature(signature, message: sighash, publicKey: publicKey) == false {
```
Comment (akifuj) [inline BitcoinKit/ScriptMachine.swift]: Is 'guard' better instead of if?

File: BitcoinKit/ScriptMachine.swift [hunk 2]
```diff
@@ -0,0 +1,882 @@
...
+
+        // Strip that last byte to have a pure signature.
+        let signature = signature.dropLast()
+
+        guard let tx = transaction else {
+            throw ScriptMachineError.error("transaction should not be nil.")
+        }
+
+        let sighash: Data = tx.signatureHash(for: utxoToSign, inputIndex: inputIndex, hashType: hashType)
+        if try Crypto.verifySignature(signature, message: sighash, publicKey: publicKey) == false {
```
Comment (akifuj) [inline BitcoinKit/ScriptMachine.swift]: Cannot you use `guard` instead of `if`

File: BitcoinKit/ScriptMachine.swift [hunk 3]
```diff
@@ -828,6 +832,10 @@ class ScriptMachine {
         // Do nothing if everything is okay.
     }
 
+    internal func testCheck(signature: Data, publicKey: Data, utxoToSign: TransactionOutput) throws {
```
Comment (akifuj) [inline BitcoinKit/ScriptMachine.swift]: What's difference from Check()? Is this any problem with access modifier?

Comment (usatie) [reply BitcoinKit/ScriptMachine.swift]: I wanted to make check() as private.
Still, I want to test this function.
http://www.mpon.me/entry/2015/05/27/100401

**Pull request merged**